src/praisonai-bot/praisonai_bot/scheduler/executor.py

```
@@ -584,6 +607,200 @@ async def _run_pinned(
+        options = dict(getattr(job, "backend_options", None) or {})
+        cwd = options.pop("cwd", None)
+
+        # Pre-run condition gate: backend turns are real (expensive) turns, so
+        # they honour the same cheap go/no-go gate as native-agent jobs. On
+        # "nothing to do" the tick is recorded ``skipped`` with no subprocess
+        # spawned; gate context is appended to the prompt like the agent path.
+        gate = self._resolve_condition(job)
+        if gate is not None:
+            try:
+                decision = await asyncio.to_thread(gate.should_run, job)
+            except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)